multinode/transaction_sender.go

```diff
@@ -201,8 +201,8 @@ func (txSender *TransactionSender[TX, RESULT, CHAIN_ID, RPC]) reportSendTxAnomal
 	}
 
 	_, criticalErr := aggregateTxResults(resultsByCode)
-	if criticalErr != nil {
-		txSender.lggr.Criticalw("observed invariant violation on SendTransaction", "tx", tx, "resultsByCode", resultsByCode, "err", criticalErr)
+	if criticalErr != nil && ctx.Err() == nil {
```

Comment:
Can't we do check similar to the one above:
```
select {
	case <-txSender.chStop:
		// it's ok to receive no results if txSender is closing. Return early to prevent false reporting of invariant violation.
		if len(resultsByCode) == 0 {
			return
		}
	default:
	}
```